src/legacy/core_plugins/table_vis/migrations.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep, get, omit } from 'lodash';
+
+export const migrations = {
+  visualization: {
+    '7.0.0': (doc) => {
+      try {
+        const visState = JSON.parse(doc.attributes.visState);
+        if (get(visState, 'type') !== 'table') {
+          return doc; // do nothing; we only want to touch tables
+        }
+
+        let splitCount = 0;
+        visState.aggs = visState.aggs.map(agg => {
+          if (agg.schema === 'split') {
+            splitCount++;
+            if (splitCount === 1) {
+              return agg; // leave the first split agg unchanged
+            }
+            agg.schema = 'bucket';
+            // the `row` param is exclusively used by split aggs, so we remove it
+            agg.params = omit(agg.params, ['row']);
+          }
+          return agg;
+        });
+
+        if (splitCount <= 1) {
+          return doc; // do nothing; we only want to touch tables with multiple split aggs
+        }
+
+        const newDoc = cloneDeep(doc);
+        newDoc.attributes.visState = JSON.stringify(visState);
+        return newDoc;
+      }
```

Comment:
I think our style guides say to put the catch on the same line as the `}` but I could be wrong? (e.g. `} catch (e) {` is how I'm used to seeing it.) Just a minor nitpick. I have prettier and tslint and eslint as plugins in VS Code, and telling VS Code to format the doc should fix all of these sorts of inconsistencies.

Reply:
Ah, thanks for the heads up. I see now that prettier puts it on one line 👍